volume/store/store.go

```diff
@@ -309,16 +309,20 @@ func (s *VolumeStore) create(name, driverName string, opts, labels map[string]st
 // This makes sure there are no races between checking for the existence of a volume and adding a reference for it
 func (s *VolumeStore) GetWithRef(name, driverName, ref string) (volume.Volume, error) {
 	name = normaliseVolumeName(name)
```

Comment:
Maybe not...
`Lock()` and `LockWithError()` cannot be mixed used for one named lock. Once the lock is created in locks[name], it should be "cancelable" or "un-cancelable".

So another bool variable, `cancelable`, needs to be added into `lockCtr` struct, then we should check that both in old `Lock()` and new `LockWithError()`.
When writing the logic, if `Lock()` meets a `cancelable` locker, the problem comes, old `Lock()` cannot return an error...

Beside, adding a new method `LockWithError()` still cannot ensure that developer will catch the error correctly and not do Unlock after `LockWithError()` returns an error.